lib/text/ui_text_displayer.js

```diff
@@ -644,30 +716,73 @@ shaka.text.UITextDisplayer = class {
       }
     }
 
-    style.lineHeight = cue.lineHeight;
+    const computedPositionAlign = this.computeCuePositionAlignment_(cue);
```

Comment:
GitHub won't let me leave comments on binary files, so here are my takes on the screenshots further down the PR.  Please correct me if I'm misinterpreting WebVTT for any of these.

 - webvtt-ui-align-center-position-10.png: I don't believe it should be squished into two lines.  I suspect that is a consequence of the width setting below.
 - webvtt-ui-align-center-position-90.png: Same.
 - webvtt-ui-align-end-long.png: This is very clearly a regression, IMO.  The text flow should not change from what it was.  The only issue with the previous version was that the background is meant to hug the text, not fill the cue box.
 - webvtt-ui-align-end.png: This should be aligned on the right (according to Chrome's native implementation), but it actually moved left of center in this PR.
 - webvtt-ui-bold-long.png and many other non-alignment tests: These all moved to the right, with the left edge centered in the screen for some reason.  I don't think this is right.

Reply:
> * webvtt-ui-align-center-position-10.png: I don't believe it should be squished into two lines.  I suspect that is a consequence of the width setting below.
> * webvtt-ui-align-center-position-90.png: Same.

Allow me to disagree. If the line is not to wrap on two lines, then what are the possibilities?
1. Display on one line and truncate the text.
2. Display the entire text on one line and don't respect the position settings.

I don't like both of these solutions and I've found 2 resources that are aligned with me (I'm not cherry-picking, I didn't find any other page):

- https://ronallo.com/demos/webvtt-cue-settings/ (drag the position cue to 10 and see what happens).
- https://www.speechpad.com/captions/webvtt
`Cue payloads are the subtitle text that is to be displayed on the screen. They come immediately after the cue timing line, and can be one or more lines in length. Line separators will be preserved. That is, even if the combined text of two lines would fit on one line, the subtitle will still be presented on two lines as indicated in the file. However, if a line will not fit the width of the cue box (aka caption box), it will be wrapped, not truncated.`

Plus if you look at example 1 at https://www.w3.org/TR/webvtt1/#introduction-multiple-lines, wrapping text on multiple lines seems to be the philosophy:
![image](https://user-images.githubusercontent.com/10497009/223676877-55add0b9-9b9c-48d6-aa0f-a5582bef525b.png)


> * webvtt-ui-align-end-long.png: This is very clearly a regression, IMO.  The text flow should not change from what it was.  The only issue with the previous version was that the background is meant to hug the text, not fill the cue box.

Again, I disagree. Without any position setting, it defaults to 50%. So the right side of the cue box should be positioned at 50% of the viewport. But I admit my implementation suffers from the same "hug" issue.


> * webvtt-ui-align-end.png: This should be aligned on the right (according to Chrome's native implementation), but it actually moved left of center in this PR.

`align:end` means that the text is right-aligned in the cue box.
Since there's no position setting, it defaults to 50.
There's no position alignment setting as well, so applying the algorithm [here](https://www.w3.org/TR/webvtt1/#webvtt-cue-position-alignment) give us the computed position alignment `line-right`.
These 3 settings gives us a cue box that is right-aligned at 50% of the viewport, with a text right-align inside it.


> * webvtt-ui-bold-long.png and many other non-alignment tests: These all moved to the right, with the left edge centered in the screen for some reason.  I don't think this is right.
Same reason as above: when no position or position alignment is given, their value are computed as described in the specs.


I think at least one of us does not understand the specs, and that could totally be me.
But reading them again and again does not make me understand them in another way.

All in all, I want to thank you for your time, and truly hope we'll finally align (right or left, whatever).